docs/source/developers/guide/step_by_step/testing.rst

```diff
@@ -29,3 +29,79 @@
 ***********
 Testing 🧪
 ***********
+
+In Arrow we use unit tests to ensure code quality, to help
+others understand our code better and to make the process
+of review easier.
+
+Adding a new unit tests is needed when adding a new feature,
+a binding or a bug fix. 
+
+It can also happen your work will involve adding a unit test
+to a code already written, modifying a unit test or even making
+more than one unit test for the changes in the code.
+
+It is also possible to make changes that do not require adding
+a unit test like :ref:`documentation`.
+
+If you do need to work with unit tests, this section will help
+you with the necessary steps.
+
+.. tabs::
+
+   .. tab:: Pytest
+
+      We use `pytest <https://docs.pytest.org/en/latest/>`_ for
+      unit tests in Python. For more info about the required
+      packages follow
+      :ref:`Python unit testing section <python-unit-testing>`.
+
+      During the development process, it can be more efficient to only run the tests relevant to the code which you are working on.  However, you should run all of the tests once you have finished your development work.
```

Comment:
On second thoughts, I wonder if this paragraph is needed either?  I don't always run all the R tests if I've made a tiny self-contained changed, as I know it's not going to impact anything else, and that the CI is going to run it all and more.  Even if this is good advice, I wonder if, given this is a step-by-step guide, it perhaps doesn't belong here.